core/dbt/task/runnable.py

```diff
@@ -375,15 +375,17 @@ def execute_nodes(self):
                     )
 
             print_run_result_error(failure.result)
-            raise
+            # ensure information about all nodes is propagated to run results when failing fast
+            return self.node_results
         except KeyboardInterrupt:
             self._cancel_connections(pool)
             print_run_end_messages(self.node_results, keyboard_interrupt=True)
             raise
-        finally:
-            pool.close()
-            pool.join()
-            return self.node_results
+
+        pool.close()
```

Comment:
Question, do we want to always do `pool.close() , pool.join()` before join? If yes, based on the document that finally will be executed before `break, return, continue`, should we consider just write it as
```python
try:
  xxxx
  return  self.node_results
except:
  xxxx
finally:
  pool.close()
  pool.join()

Reply:
That may be a very sound thing to do, but I think the safest path forward to ensure the regression (and any others that may be lingering) are fixed is to go back to the state before this change: https://github.com/dbt-labs/dbt-core/pull/8066/files#diff-f01af78b7eaa1bacc301afa75eea4c7cdbe03064b3913bc2ce3f6bf01d0bc125R383-R386, where there was no `finally`